datastore.go

```diff
@@ -337,12 +313,12 @@ func (d *Datastore) Close() error {
 	d.closeOnce.Do(func() {
 		close(d.closing)
```

Comment:
We can safety move this after the SetToIf check and remove the Once.Do wrapper.

Reply:
good call